watertap/costing/watertap_costing_package.py

```diff
@@ -61,23 +63,213 @@ def add_LCOW(self, flow_rate, name="LCOW"):
+            agg_fixed_opex_lcows[unit_model_class] += fixed_opex_lcows[
+                u.unit_model.name
+            ]
+            agg_variable_opex_lcows[unit_model_class] += variable_opex_lcows[
+                u.unit_model.name
+            ]
+
+        for ftype, flows in self._registered_flows.items():
+            # part of total_variable_operating_cost
+            if ftype in agg_variable_opex_lcows:
+                raise RuntimeError(
+                    f"Found unit model named {ftype} but want to name flow {ftype} for {name+'_component_variable_opex'}"
+                )
+            if ftype not in self.used_flows:
+                assert len(flows) == 0
+                continue
+            agg_variable_opex_lcows[ftype] = (
+                self.aggregate_flow_costs[ftype] * self.utilization_factor / denominator
+            )
+            cost_var = getattr(self, f"{ftype}_cost")
+            for flow_expr in flows:
+                # part of total_variable_operating_cost
+                flow_cost = pyo.units.convert(
+                    flow_expr * cost_var, to_units=c_units / t_units
+                )
+                unit_name = self._find_flow_unit_name(flow_expr, variable_opex_lcows)
+                if unit_name is not None:
+                    variable_opex_lcows[unit_name] += (
+                        flow_cost * self.utilization_factor
+                    ) / denominator
+                    continue
+                flow_name = self._get_flow_name(flow_expr)
+                if flow_name in variable_opex_lcows:
+                    raise RuntimeError(
+                        f"Found unit model named {flow_name} but need to name flow {flow} for {name+'_aggregate_variable_opex'}"
+                    )
+                variable_opex_lcows[flow_name] = (
+                    flow_cost * self.utilization_factor
+                ) / denominator
+
+    @staticmethod
+    def _get_flow_expr_var(flow_expr):
+        """
+        Try to find the variable associated with the flow flow_expr.
+        This will attempt to get the specific variable determining the flow.
+        """
+        variables = list(identify_variables(flow_expr, include_fixed=True))
+        if len(variables) == 1:
+            # only one variable, assume it's the variable controlling this flow
+            return variables[0]
+        variables = list(identify_variables(flow_expr, include_fixed=False))
+        if len(variables) == 1:
+            # only one variable, assume it's the variable controlling this flow
+            return variables[0]
+        # can't be sure which variable controls the flow, so return None
+        return None
+
+    def _find_flow_unit_name(self, flow_expr, variable_opex_lcows=None):
+        """
+        Try to find a unit already in variable_opex_lcows to which `flow_expr` is attached.
+        """
+        variable = self._get_flow_expr_var(flow_expr)
+        if variable is None:
+            return None
```

Comment:
It doesn't matter if there's only one variable in the flow expression, just that the variables in the flow expression are attached to the same unit.

This would allow us to identify the unit for more general expressions.